cypress/e2e/permit2.test.ts

```diff
@@ -0,0 +1,206 @@
+import { MaxUint160, MaxUint256 } from '@uniswap/permit2-sdk'
+
+import { DAI, USDC_MAINNET } from '../../src/constants/tokens'
+
+const APPROVE_BUTTON = '[data-testid="swap-approve-button"]'
+
+/** Initiates a swap and confirms its success. */
+const swaps = () => {
+  // Ensures that the swap button is enabled, since it can be temporarily disabled following approval.
+  cy.get('#swap-button').should('not.have.attr', 'disabled')
+
+  // Completes the swap.
+  cy.get('#swap-button').click()
+  cy.get('#confirm-swap-or-send').click()
+  cy.get('[data-testid="dismiss-tx-confirmation"]').click()
+
+  // Verifies that there is a successful swap notification.
+  cy.contains('Swapped').should('exist')
+}
+
+// TODO: Update tests to differentiate between permit2 vs token approval button text once UI is updated to indicate approval step.
```

Comment:
Any TODO needs a linked jira ticket.
If there's already a ticket for the button text disambiguation, can you re-use that ticket number?

Reply:
yes was waiting to see if there was an existing ticket but there wasn't so I made one